Add GCP PubSub configuration

jlestel opened the pull request:
As a GCP user, I want to use GCP PubSub with my Telemetry instance.

jlestel committed:
Update values.yaml

Commit:
up

Commit:
feat: Add dynamic env

Commit:
add env

Commit:
Merge branch 'main' of https://github.com/MyTeslaMate/helm-charts

Commit:
up

jlestel committed:
Update values.yaml

jlestel committed:
Update Chart.yaml

jlestel committed:
Update README.md

agbpatro commented on charts/fleet-telemetry/templates/2-deployment.yaml:
```diff
@@ -27,6 +27,10 @@ spec:
           value: {{ quote .Values.kubernetesClusterDomain }}
+        {{- if .Values.gcpSecret }}
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /etc/gcp/key.json
```
shouldn't this be `/etc/gcp/{{.Values.gcpSecret}}`
I dont know the proper syntax for it

jlestel committed:
Add comment

jlestel commented on charts/fleet-telemetry/values.yaml:
```diff
@@ -39,3 +39,4 @@ config:
   profile:
     port: 4269
 kubernetesClusterDomain: cluster.local
+gcpSecret: null
```
This is to inform of the presence of the variable. I added a comment above to describe how it works.
Thus, if we generate the associated documentation this optional parameter will be mentioned.

jlestel committed:
Remove useless default value

jlestel committed:
revert

jlestel committed:
Merge branch 'teslamotors:main' into main

jlestel committed:
Merge branch 'teslamotors:main' into main

Commit:
Add checksum annotations for config and secret in deployment template

Commit:
fix(fleet-telemetry): TCP liveness on the TLS port, not /metrics

fleet-telemetry >= v0.9 no longer serves prometheus metrics on
config.metrics.port (9273). The liveness probe still did httpGet /metrics
there, so it failed every ~50s and crashlooped the pod (observed: 184
restarts/10h on v0.9.3), which repeatedly dropped the vehicle stream and
backed up the Pub/Sub pipeline (slow acks, alert spikes).

Switch liveness to a tcpSocket check on the main TLS port (config.port) —
"is the server accepting vehicle connections". Readiness already uses the
status server (/status on config.status.port), so it's unaffected.

Bump chart to 0.1.2.

jlestel committed:
Merge pull request #1 from MyTeslaMate/fix/fleet-telemetry-liveness-probe

fix(fleet-telemetry): TCP liveness on the TLS port, not /metrics

Commit:
Revert liveness to httpGet /metrics now that metrics bind on 0.0.0.0

The tcpSocket workaround (helm-charts#1) was needed only because fleet-telemetry
v0.9 defaulted the prometheus metrics endpoint to 127.0.0.1, so probing
pod-IP:9273 failed. With prometheus_metrics_host set to 0.0.0.0 (app side),
9273 is reachable on the pod IP again (verified: OPEN + /metrics returns data).
Restore the original httpGet /metrics liveness — a real health check, no TLS
handshake noise. readinessProbe (/status) is unchanged. Bump chart to 0.1.3.

jlestel committed:
Merge pull request #2 from MyTeslaMate/fix/fleet-telemetry-restore-metrics-liveness

Revert liveness to httpGet /metrics now that metrics bind on 0.0.0.0
